src/microcosm.js

```diff
@@ -206,24 +206,38 @@ class Microcosm extends Emitter implements Domain {
    * that, when rolling back to this action, it always has a state value.
    */
   createSnapshot(action: Action) {
-    this.snapshots[action.id] = this.recall(action.parent)
+    this.snapshots[action.id] = {
+      last: this.state,
+      next: this.state,
+      status: action.status,
+      payload: action.payload
+    }
   }
 
   /**
    * Update the state snapshot for a given action
    */
   updateSnapshot(action: Action) {
-    let next = this.recall(action.parent)
+    // Fall back to creating a snapshot if it does not exist in the event
+    // an action is in progress while a fork is being created
+    let snap = this.snapshots[action.id]
+    let last = this.recall(action.parent)
 
     if (this.parent) {
-      next = merge(next, this.parent.recall(action))
+      last = merge(last, this.parent.recall(action))
     }
```

Comment:
Excellent question.

Right now, parents always update their snapshots before their children because they are created first, so they subscribe to history first. This means they get executed first on the list of callbacks whenever history emits update.

So we don't have to do this. But I want to give it a shot. I think it might avoid some edge cases in the future if we start exploring that.